src/Utilities/FlowAnalysis/Analysis/TaintedDataAnalysis/TaintedDataAnalysis.TaintedDataOperationVisitor.cs

```diff
@@ -0,0 +1,472 @@
+                    && !subResult.TaintedDataSourceSinks.IsEmpty)
+                {
+                    foreach (TaintedDataSourceSink sourceSink in subResult.TaintedDataSourceSinks)
+                    {
+                        if (!this.TaintedSourcesBySink.TryGetValue(sourceSink.Sink, out HashSet<SymbolAccess> sourceOrigins))
+                        {
+                            sourceOrigins = new HashSet<SymbolAccess>();
+                            this.TaintedSourcesBySink.Add(sourceSink.Sink, sourceOrigins);
+                        }
+
+                        sourceOrigins.UnionWith(sourceSink.SourceOrigins);
+                    }
+                }
+            }
+
+            private void ProcessAssignmentOperation(IAssignmentOperation assignmentOperation)
+            {
+                TaintedDataAbstractValue assignmentValueAbstractValue = this.GetCachedAbstractValue(assignmentOperation.Value);
+                if (assignmentOperation.Target != null
+                    && assignmentValueAbstractValue.Kind == TaintedDataAbstractValueKind.Tainted
+                    && assignmentOperation.Target is IPropertyReferenceOperation propertyReferenceOperation
+                    && this.IsPropertyASink(propertyReferenceOperation))
+                {
+                    this.TrackTaintedDataEnteringSink(propertyReferenceOperation.Member, propertyReferenceOperation.Syntax.GetLocation(), assignmentValueAbstractValue.SourceOrigins);
+                }
+            }
+
+            private IEnumerable<TaintedDataAbstractValue> GetTaintedValuesFromInputArguments(IEnumerable<IArgumentOperation> arguments)
+            {
+                foreach (IArgumentOperation argument in arguments)
+                {
+                    if (argument.Parameter.RefKind == RefKind.None
+                        || argument.Parameter.RefKind == RefKind.Ref
+                        || argument.Parameter.RefKind == RefKind.In)
+                    {
+                        TaintedDataAbstractValue value = this.GetCachedAbstractValue(argument);
+                        if (value.Kind == TaintedDataAbstractValueKind.Tainted)
+                        {
+                            yield return value;
+                        }
+                    }
+                }
+            }
+
+            /// <summary>
+            /// Determines if the instance method call returns tainted data.
+            /// </summary>
+            /// <param name="wellKnownTypeProvider">Well known types cache.</param>
+            /// <param name="method">Instance method being called.</param>
+            /// <returns>True if the method returns tainted data, false otherwise.</returns>
+            private bool IsSanitizingMethod(IMethodSymbol method)
+            {
+                foreach (SanitizerInfo sanitizerInfo in this.DataFlowAnalysisContext.SanitizerInfos.GetInfosForType(method.ContainingType))
```

Comment:
>GetInfosForType [](start = 100, length = 15)

Ah, sorry did not see this callsite of `GetInfosForType`. Probably ignore some of my earlier comments about it...

Reply:
Or wait, this is SanitizerInfos, not SourceInfos correct? Your call - consider caching the relevant information (set of infos or set of tained symbols) for hot code paths.

---
In reply to: [234018680](https://github.com/dotnet/roslyn-analyzers/pull/1882#discussion_r234018680) [](ancestors = 234018680)

Reply:
I'll open an issue to consider it.  I think I'll want to collect some performance data first.